bundles/org.openhab.binding.elroconnects/src/main/java/org/openhab/binding/elroconnects/internal/handler/ElroConnectsAccountHandler.java

```diff
@@ -156,71 +167,81 @@ private void stopPolling() {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                "@text/offline.request-failed");
+                        return null;
+                    }
+                    Map<String, ElroConnectsConnector> devices = new HashMap<>();
+                    if (response != null) {
+                        response.forEach(d -> devices.put(d.getDevTid(), d));
+                    }
+                    this.devices = devices;
+                    updateStatus(ThingStatus.ONLINE);
                 } else {
-                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
-                            accountException.getLocalizedMessage());
+                    if (accountException == null) {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                    } else {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                accountException.getLocalizedMessage());
+                    }
                 }
-            }
 
-            return null;
-        });
+                return null;
+            }).get();
+        } catch (InterruptedException e) {
+            updateStatus(ThingStatus.OFFLINE);
+        } catch (ExecutionException e) {
+            logger.debug("Poll exception", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+        }
     }
 
     private void login() {
         logger.debug("Login");
-        getAccessToken().handle((accessToken, accountException) -> {
-            this.accessToken = accessToken;
-            if (accessToken != null) {
-                updateStatus(ThingStatus.ONLINE);
-            } else {
-                if (accountException == null) {
-                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+        try {
+            getAccessToken().handle((accessToken, accountException) -> {
+                this.accessToken = accessToken;
+                if (accessToken != null) {
+                    updateStatus(ThingStatus.ONLINE);
                 } else {
-                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
-                            accountException.getLocalizedMessage());
+                    if (accountException == null) {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                    } else {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                accountException.getLocalizedMessage());
+                    }
                 }
-            }
 
-            return null;
-        });
+                return null;
+            }).get();
+        } catch (InterruptedException e) {
+            updateStatus(ThingStatus.OFFLINE);
```

Comment:
I think you should call `Thread.currentThread().interrupt();` in this case.

Reply:
Done